requirements.txt

```diff
@@ -13,3 +13,4 @@ SQLAlchemy==1.3.16
 strict-rfc3339==0.7
 -e git+https://github.com/tl-its-umich-edu/api-utils-python@v1.3#egg=umich-api
 yoyo-migrations==7.0.2
+ptvsd>=4.3.2,<4.3.99
```

Comment:
With all other dependencies we've pinned to the exact bugfix release. Was the range here intentional? Should we be using ranges elsewhere?

Reply:
Also, I will politely request that this be put in the proper place to maintain alphabetical order.

Reply:
Yeah, I just copied from MyLA, I can lock the version.

They haven't released a new version of this package since August 2019 but if this isn't updated together when VSCode is updated, generally has broken compatibility with VSCode. But we can update that when it happens.

Reply:
I think the range for this library should be fine here since this is purely for local development and we want to update the library automatically to support a new version of VSCode  released.

Reply:
I know we've gone back and forth about this with our projects. I found this fairly lucid StackOverflow answer that I thought might be worth sharing: https://stackoverflow.com/a/44938662

It seems they're suggesting using ranges in PyPi packages and pinning dependencies for custom applications.

Reply:
I'd personally prefer we have every version locked, including the versions of packages installed in apt. This should make the builds consistently reproducible forever.

But the problem is that it requires effort to have a consistent release schedule that also includes reviewing and updating all of the dependencies. (Or like the SO says have a service that reminds us to do that)

I've seen as a project gets bigger with less focus, dependency updating usually is forgotten. :( It would be nice to improve on that, but it also hasn't caused any problems yet.

Reply:
Yeah, I think I agree. I think having someone do research on the best services for this would be great, would help us for many projects. In the meantime, you just have to do a little legwork, and I still think we should be doing this for each minor or major release cycle. Note that I did do for 2020.01.01 (in PR #140). It doesn't take that long, especially when we only have 15ish dependencies.